contracts/mainnet/connectors/mstable/main.sol

```diff
@@ -0,0 +1,363 @@
+pragma solidity ^0.7.6;
+
+/**
+ * @title mStable SAVE.
+ * @dev Depositing and withdrawing directly to Save
+ */
+
+import { Helpers } from "./helpers.sol";
+import { Events } from "./events.sol";
+import { IMasset, ISavingsContractV2, IBoostedSavingsVault, IFeederPool } from "./interface.sol";
+import { TokenInterface } from "../../common/interfaces.sol";
+
+abstract contract mStableResolver is Events, Helpers {
+	//
+	/***************************************
+                    CORE
+    ****************************************/
+
+	/**
+	 * @dev Deposit to Save via mUSD or bAsset
+	 * @notice Deposits token supported by mStable to Save
+	 * @param _token Address of token to deposit
+	 * @param _amount Amount of token to deposit
+	 * @param _minOut Minimum amount of token to mint/deposit, equal to _amount if mUSD
+	 * @return _eventName Event name
+	 * @return _eventParam Event parameters
+	 */
+
+	function deposit(
+		address _token,
+		uint256 _amount,
+		uint256 _minOut
+	) external returns (string memory _eventName, bytes memory _eventParam) {
+		//
+		uint256 mintedAmount = _amount;
+
+		// Check if needs to be minted first
+		if (IMasset(mUsdToken).bAssetIndexes(_token) != 0) {
+			// mint first
+			approve(TokenInterface(_token), mUsdToken, _amount);
+			mintedAmount = IMasset(mUsdToken).mint(
+				_token,
+				_amount,
+				_minOut,
+				address(this)
+			);
+		} else {
+			require(mintedAmount >= _minOut, "mintedAmount < _minOut");
+		}
+
+		return _deposit(_token, mintedAmount, imUsdToken);
+	}
+
+	/**
+	 * @dev Deposit to Save via feeder pool
+	 * @notice Deposits token, requires _minOut for minting and _path
+	 * @param _token Address of token to deposit
+	 * @param _amount Amount of token to deposit
+	 * @param _minOut Minimum amount of token to mint
+	 * @param _path Feeder Pool address for _token
+	 * @return _eventName Event name
+	 * @return _eventParam Event parameters
+	 */
+
+	function depositViaSwap(
+		address _token,
+		uint256 _amount,
+		uint256 _minOut,
+		address _path
+	) external returns (string memory _eventName, bytes memory _eventParam) {
+		//
+		require(_path != address(0), "Path must be set");
+		require(
+			IMasset(mUsdToken).bAssetIndexes(_token) == 0,
+			"Token is bAsset"
+		);
+
+		approve(TokenInterface(_token), _path, _amount);
+		uint256 mintedAmount = IFeederPool(_path).swap(
+			_token,
+			mUsdToken,
+			_amount,
+			_minOut,
+			address(this)
+		);
```

Comment:
Here, the event should be emitted!

Reply:
the _deposit functions returns the event data. Does that work?

Reply:
Totally